static/src/javascripts/projects/common/modules/experiments/ab-ophan.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+
+import type {
+    ABTest,
+    Variant,
+    OphanABEvent,
+    OphanABPayload,
+} from 'common/modules/experiments/ab-types';
+
+import { getActiveTests } from 'common/modules/experiments/ab-tests';
+import * as testCanRunChecks
+    from 'common/modules/experiments/test-can-run-checks';
+import * as segmentUtil from 'common/modules/experiments/segment-util';
+import * as abUtils from 'common/modules/experiments/utils';
+import config from 'lib/config';
+import reportError from 'lib/report-error';
+import ophan from 'ophan/ng';
+
+const not = f => (...args): boolean => !f.apply(this, args);
+const noop = (): null => null;
+
+const submit = (payload: OphanABPayload): void =>
+    ophan.record({
+        abTestRegister: payload,
+    });
+
+/**
+ * generate an A/B event for ophan
+ */
+const makeABEvent = (
+    variant: Variant,
+    complete: string | boolean
+): OphanABEvent => {
+    const event: OphanABEvent = {
+        variantName: variant.id,
+        complete,
+    };
+
+    if (variant.options && variant.options.campaignCodes) {
+        event.campaignCodes = variant.options.campaignCodes;
+    }
+
+    return event;
+};
+
+/**
+ * Checks if this test will defer its impression by providing its own impression function.
+ *
+ * If it does, the test won't be included in the Ophan call that happens at pageload, and must fire the impression
+ * itself via the callback passed to the `impression` property in the variant.
+ *
+ * @param test
+ * @returns {boolean}
+ */
```

Comment:
We probably don't need these fancy-pants `@param` / `@return` type comments, now that we have The Real Deal:tm: